py/server/deephaven/completer/__init__.py

```diff
@@ -0,0 +1,95 @@
+#
+# Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+#
+
+""" This module allows the user to configure if and how we use jedi to perform autocompletion.
+See https://github.com/davidhalter/jedi for information on jedi.
+
+# To disable autocompletion
+from deephaven.completer import CompleterSettings, CompleterMode
+CompletionSettings.mode = 'off'
+
+Valid options for completer_mode are one of: [off, safe, strong].
+off: do not use any autocomplete
+safe mode: uses static analysis of source files. Can't execute any code.
+strong mode: looks in your globals() for answers to autocomplete and analyzes your runtime python objects
+later, we may add slow mode, which uses both static and interpreted completion modes.
+"""
+
+from enum import Enum
+
+
+class CompleterMode(Enum):
+    off = 'off'
+    safe = 'safe'
+    strong = 'strong'
+
+    def __str__(self) -> str:
+        return self.value
+
+
+class Completer(object):
+
```

Comment:
This seems to be a 'private' class for creating a singleton, if so, change it to _Completer ?

Reply:
Java actually holds onto this object and calls into its methods, but the consumer would probably only care about `jedi_settings` and not `Completer`.  Will rename it.

Reply:
Still need direction on this.

Also, should I move it to its own file, or leave things in `__init__.py`.  Most others, init seems very sparse, if you'd rather structure it differently, please lmk

Reply:
If we don't think there will likely be more modules in the completer sub-package, I would just have _completer.py and keep the class name as it is.

Reply:
done